src/auth/auth_azure_ad_oauth.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+import tornado.auth
+
+from auth.auth_abstract_oauth import AbstractOauthAuthenticator, _OauthUserInfo
+from model import model_helper
+
+LOGGER = logging.getLogger('script_server.AzureADOauthAuthenticator')
+
+
+class AzureAdOAuthAuthenticator(AbstractOauthAuthenticator):
+    def __init__(self, params_dict):
+        params_dict['group_support'] = False
+        self.auth_url = model_helper.read_obligatory(params_dict, 'auth_url', ' for OAuth')
```

Comment:
May be we should ask only for `tenant_id` from the config? Since the URLs always have the same structure

Reply:
I believe these URLs can be copy-pasted directly from the Azure Portal when a new app is registered. So I guess to semi-future proof it, we can leave the full URL in the config in case Microsoft changes the URL structure.

For reference, I was mostly going by the [Grafana Azure AD setup instructions](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/azuread/#create-the-microsoft-entra-id-application)

Reply:
ok, fair, thanks!